Clarify what to do when ALPN negotiation fails (#2483)
quicwg/base-drafts: draft-ietf-quic-tls.md

{{!RFC7301}} to select an application protocol.  Unless another mechanism is
used for agreeing on an application protocol, endpoints MUST use ALPN for this

> purpose.  When using ALPN, endpoints MUST abort a connection if an application

Reviewer: This isn't within your change, but what  does "abort" mean. Should this be "drop QUIC connection state"?

Reviewer 2: I think that an immediate close is what was intended.

Reviewer: ```suggestion
purpose.  When using ALPN, endpoints MUST immediately close a connection (see Section 10.3 in {{QUIC-TRANSPORT}}) if an application
```

> protocol is not negotiated with a no_application_protocol alert (error code

Reviewer: ```suggestion
protocol is not negotiated with a no_application_protocol TLS alert (QUIC error code
```

> 0x178). While {{!RFC7301}} only specifies that servers use this alert, QUIC

Reviewer: ```suggestion
0x178, see {{tls-errors}}).  While {{!RFC7301}} only specifies that servers use this alert, QUIC
```
